### ISSUE TITLE
Section titles in M2.7+

### ISSUE THREAD

**Issue by gjb2048** [gjb2048/moodle-format_topcoll#21]
Hi... We're running Moodle 2.7.11+ (2014051211.01), and have been running the Collapsible Topics format a few months without any problems. We installed the latest version -- 2.7.4 (2014092005) for Moodle 2.7 -- on our test server, and now notice that section titles do not display in courses with using this format (new course, too). When we revert back to the previous version, the section titles come back. We are running PHP version 5.5.29. I also installed a fresh Moodle 2.7.11 version on my local machine and installed the collapsible formats topic, and notice the same problem. 

Is there something else that we need to do for the update to work properly? Or perhaps is this a problem others have seen?

Thanks much..... Brian

**Comment by gjb2048** [comment]
And in M3.0 too.

**Comment by gjb2048** [comment]
Cannot currently replicate on M2.7 or M3.0 - awaiting info:

Hi Brian,

The section titles are working for me in both the M2.7 and M3.0 versions - they are the toggle headings after all.  Please post screen-shots on: https://moodle.org/mod/forum/view.php?id=47 with full details of the theme you are using.

Kind regards,

Gareth

**Comment by gjb2048** [comment]
Not a bug....

Hi Gareth --

When we installed the Collapsed Topics format a couple months ago, we changed the following options to #FFFFFF (sorry if I wasn't clear here earlier):

Toggle foreground color
format_topcoll | defaulttgfgcolour

Toggle foreground hover colour
format_topcoll | defaulttgfghvrcolour

Toggle background color
format_topcoll | defaulttgbgcolour

Toggle background hover color
format_topcoll | defaulttgbghvrcolour

And Toggle all icon hovers was set to YES

---

I see what you mean when you asked why we set them all to white. That doesn't make sense. Even though these are set to white, when running the previous version of Collapsed Topics, the section titles are actually not white (that's probably why we didn't dwell on the default color settings). They are using the color from our custom theme (note that if I switch the theme to the Clean theme, the titles do go white; when I switch back to our custom theme, the titles are visible again, using our the color from our theme). It's only after installing the updated Collapsed Topics do the titles go to white. 

I think it's interesting, using the previous version and before installing the update, the section title color was essentially ignoring the course format's settings and instead used our custom theme's color setting. It's only after installing the upgrade does it all sort itself out.... the format's color settings are respected. So I suppose there isn't anything necessarily wrong with the course format. We will have to update the default color settings (i.e. don't use white for text).

Does this make sense? I apologize if I led you down the wrong path, trying to find an error where one doesn't exist.

Thanks... Brian

Hi Brian,

Thank you for the information.

A recent fix: https://github.com/gjb2048/moodle-format_topcoll/commit/3cd8a1d0641419158d76a11a6725329d7e1a6b02 caused the incorrect data to manifest itself.

Kind regards,

Gareth
